daemon/api_aspects.go

```diff
@@ -0,0 +1,118 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+/*
+ * Copyright (C) 2023 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package daemon
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/snapcore/snapd/aspects"
+	"github.com/snapcore/snapd/overlord/auth"
+)
+
+var (
+	aspectsCmd = &Command{
+		Path:        "/v2/aspects",
+		GET:         getAspect,
+		POST:        setAspect,
+		ReadAccess:  openAccess{},
+		WriteAccess: authenticatedAccess{Polkit: polkitActionManage},
+	}
+)
+
+type AspectRequest struct {
+	Account    string      `json:"account"`
+	BundleName string      `json:"bundle"`
+	Aspect     string      `json:"aspect"`
+	Field      string      `json:"field"`
+	Value      interface{} `json:"value"`
+}
+
+const emptyFieldFmt = "cannot have empty %q field"
+
+func (r *AspectRequest) validate() error {
```

Comment:
This validation made me consider possible validation requirements and impact of where it is directly or indirectly checked. It seems that we would want to keep databag retrievals to a minimum and therefore validate as much as possible around the validity of the aspect path (account/bundle/aspect/field) before a data bundle retrieval happens. 
Are any format requirements for "aspect path" that should/can be checked directly before. Is caching of aspect paths or some technique for directly validating existence of entries perhaps a consideration to help minimise access requirements?